### PR TITLE
Feature/get arch adds an empty archive job to GEFS system

### PR DESCRIPTION
**Pull request by AntonMFernando-NOAA**
# Description

- This feature added an archive task for GEFS system. This task will run in the service queue with a single processor. 
The task executes with success.
- Only xml is generated.

  Resolves #
  Refs #832 #2698
  Refs NOAA-EMC/repo#

# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

- Forecast-only on Hercules

# Checklist
- [ ] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: workflow/rocoto/gefs_tasks.py [hunk 1]
```diff
@@ -483,3 +483,25 @@ def wavepostpnt(self):
         task = rocoto.create_task(member_metatask_dict)
 
         return task
+    
+    def arch(self):
+        deps = []
+        dependencies = []
+        dep_dict = {'type': 'task', 'name': f'stage_ic'}
```
Comment (aerorahul) [inline workflow/rocoto/gefs_tasks.py]: This task should depend on the successful completion of the post-processing tasks.
For reference, take a look at `def arch` in `gfs_tasks.py` for constructing a multi-dependency  list.

File: workflow/rocoto/gefs_tasks.py [hunk 2]
```diff
@@ -483,3 +483,25 @@ def wavepostpnt(self):
...
+        dependencies.append(rocoto.add_dependency(dep_dict))
+        dependencies = rocoto.create_dependency(dep_condition='and', dep=deps)
+        resources = self.get_resource('arch')
+        task_name = f'arch'
+        task_dict = {'task_name': task_name,
+                     'resources': resources,
+                     'envars': None,
```
Comment (aerorahul) [inline workflow/rocoto/gefs_tasks.py]: Refer `gfs_tasks.py`

File: parm/config/gefs/config.resources [hunk 3]
```diff
@@ -293,6 +293,17 @@ case ${step} in
...
 
+  "arch")
+    declare -x "wtime_${step}"="06:00:00"
+    declare -x "npe_${step}"="1"
+    declare -x "npe_node_${step}"="1"
+    declare -x "nth_${step}"="1"
+    declare -x "memory_${step}"="4096M"
+    if [[ "${machine}" == "WCOSS2" ]]; then
+      declare -x "memory_${step}"="50GB"
+    fi
```
Comment (aerorahul) [inline parm/config/gefs/config.resources]: This section needs to be in `config.resources.WCOSS2` 
Take a look at the `config.resources` in `gfs` for example.

**Review by aerorahul — CHANGES_REQUESTED** [review]
A few comments that need attention.

File: parm/config/gefs/config.resources (hunk 3 above)
Comment (AntonMFernando-NOAA) [reply parm/config/gefs/config.resources]:
```
    if [[ "${machine}" == "WCOSS2" ]]; then
      declare -x "memory_${step}"="50GB"
    fi
```
This part is removed

File: workflow/rocoto/gefs_tasks.py [hunk 4]
```diff
@@ -483,3 +483,45 @@ def wavepostpnt(self):
         task = rocoto.create_task(member_metatask_dict)
 
         return task
+
+    def arch(self):
+        deps = []
+        dep_dict = {'type': 'metatask', 'name': f'atmos_prod'}
```
Comment (aerorahul) [inline workflow/rocoto/gefs_tasks.py]:
```suggestion
        dep_dict = {'type': 'metatask', 'name': 'atmos_prod'}
```
`f` is used when there is something to substitute.  It is unnecessary here and in the rest of the dependencies.

File: workflow/rocoto/gefs_tasks.py [hunk 5]
```diff
@@ -483,3 +483,45 @@ def wavepostpnt(self):
...
+            deps.append(rocoto.add_dependency(dep_dict))
+        dep_dict = {'type': 'metatask', 'name': f'wave_post_grid'}
+        deps.append(rocoto.add_dependency(dep_dict))
+        dependencies = rocoto.create_dependency(dep=deps, dep_condition='and')
+
+        resources = self.get_resource('arch')
+        task_name = f'arch'
+        task_dict = {'task_name': task_name,
+                     'resources': resources,
+                     'envars': None,
```
Comment (aerorahul) [inline workflow/rocoto/gefs_tasks.py]: `envars` are required.

Comment (AntonMFernando-NOAA) [reply workflow/rocoto/gefs_tasks.py]: added `'envars': self.envars`

File: workflow/rocoto/gefs_tasks.py (hunk 4 above)
Comment (AntonMFernando-NOAA) [reply workflow/rocoto/gefs_tasks.py]: fixed

File: workflow/rocoto/gefs_tasks.py (hunk 1 above)
Comment (AntonMFernando-NOAA) [reply workflow/rocoto/gefs_tasks.py]: @aerorahul with this part of the code the arch job doesn't pass. See any issue?
```
        if self.app_config.do_wave:
            dep_dict = {'type': 'task', 'name': 'wavepostsbs'}
            deps.append(rocoto.add_dependency(dep_dict))
            if self.app_config.do_wave_bnd:
                dep_dict = {'type': 'task', 'name': 'wavepostbndpnt'}
                deps.append(rocoto.add_dependency(dep_dict))
                dep_dict = {'type': 'task', 'name': 'wavepostbndpntbll'}
                deps.append(rocoto.add_dependency(dep_dict))
            dep_dict = {'type': 'task', 'name': 'wavepostpnt'}
            deps.append(rocoto.add_dependency(dep_dict))
```

File: workflow/rocoto/gefs_tasks.py [hunk 6]
```diff
@@ -483,3 +483,45 @@ def wavepostpnt(self):
...
+            dep_dict = {'type': 'task', 'name': 'wavepostsbs'}
+            deps.append(rocoto.add_dependency(dep_dict))
+            if self.app_config.do_wave_bnd:
+                dep_dict = {'type': 'task', 'name': 'wavepostbndpnt'}
+                deps.append(rocoto.add_dependency(dep_dict))
+                dep_dict = {'type': 'task', 'name': 'wavepostbndpntbll'}
+                deps.append(rocoto.add_dependency(dep_dict))
+            dep_dict = {'type': 'task', 'name': 'wavepostpnt'}
+            deps.append(rocoto.add_dependency(dep_dict))
+        dep_dict = {'type': 'metatask', 'name': 'wave_post_grid'}
```
Comment (aerorahul) [inline workflow/rocoto/gefs_tasks.py]: shouldn't this be indented under the `if self.app_config.do_wave:` condition?

**Review by aerorahul — DISMISSED** [review]
looks good.

File: jobs/rocoto/arch_test.sh [hunk 7]
```diff
@@ -0,0 +1,8 @@
+#! /usr/bin/env bash
+
+###############################################################
+# Execute the JJOB
+echo "test is successful!"
+status=$?
```
[the remark this answers is not in the document]
Comment (AntonMFernando-NOAA) [reply jobs/rocoto/arch_test.sh]: This will be fixed in the next pull request

**Comment by AntonMFernando-NOAA** [comment]
@aerorahul I think now the pull request is ready to merge. Please advice.

**Comment by WalterKolczynski-NOAA** [comment]
@AntonMFernando-NOAA The pynorms and shellnorms failures need to be corrected before it can be merged.

**Review by aerorahul — DISMISSED** [review]
lgtm.

**Comment by AntonMFernando-NOAA** [comment]
@WalterKolczynski-NOAA 



> @AntonMFernando-NOAA The pynorms and shellnorms failures need to be corrected before it can be merged.

@WalterKolczynski-NOAA Now all the checks have passed. I think now the PR is ready to merge.

**Review by WalterKolczynski-NOAA — DISMISSED** [review]
Conditionally approved.

**Comment by WalterKolczynski-NOAA** [comment]
@AntonMFernando-NOAA `arch_test.sh` doesn't appear to have executable permissions.

**Comment by AntonMFernando-NOAA** [comment]
> @AntonMFernando-NOAA `arch_test.sh` doesn't appear to have executable permissions.

@WalterKolczynski-NOAA I think I fixed it. However, permission change to a file cannot be added as a commit in git. Please check whether the ci test passes.

**Comment by WalterKolczynski-NOAA** [comment]
> > @AntonMFernando-NOAA `arch_test.sh` doesn't appear to have executable permissions.
> 
> @WalterKolczynski-NOAA I think I fixed it. However, permission change to a file cannot be added as a commit in git. Please check whether the ci test passes.

You _should_ be able to.

**Comment by DavidHuber-NOAA** [comment]
@AntonMFernando-NOAA The mode still is not executable.  This worked for me:
```console
> cd jobs/rocoto
> chmod +x arch_test.sh
> git diff arch_test.sh
diff --git a/jobs/rocoto/arch_test.sh b/jobs/rocoto/arch_test.sh
old mode 100644
new mode 100755
> git commit arch_test.sh "Make arch_test.sh executable"
```

**Comment by AntonMFernando-NOAA** [comment]
> @AntonMFernando-NOAA The mode still is not executable. This worked for me:
> 
> ```
> > cd jobs/rocoto
> > chmod +x arch_test.sh
> > git diff arch_test.sh
> diff --git a/jobs/rocoto/arch_test.sh b/jobs/rocoto/arch_test.sh
> old mode 100644
> new mode 100755
> > git commit arch_test.sh "Make arch_test.sh executable"
> ```

@WalterKolczynski-NOAA @DavidHuber-NOAA Good to go now.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Review by DavidHuber-NOAA — APPROVED** [review]

**Pull request merged**